File: ProductExtractor/repository/provider_repository.py
from repository.repository import Repository
from repository.utils import insert, select

from models.provider import Provider


class ProviderRepository(Repository[Provider]):
    @classmethod
    def tablename(cls) -> str:
        return "Provider"

    @classmethod
    def add(cls, item: Provider) -> int:
        rows = select(cls.tablename(), ["ID"], {"Name": item.name})
        if len(rows) > 0:
            return rows[0][0]

        return insert(
            ProviderRepository.tablename(),
            {
                "Name": item.name,
                "Icon": item.icon,
                "Link": item.link,
            },
        )

File: src/Repositories/ProviderRepository.php
<?php

namespace App\Repositories;
use App\Entities\Provider;

use Exception;

class ProviderRepository extends Repository {
    protected static string $tableName = "Provider";

    public static function getByID(int $provider_id): Provider {
        $data = self::select(['ID' => $provider_id])[0];
        return self::convertToProvider($data);
    }

    private static function convertToProvider(object $data): Provider {
        return new Provider(
            $data->ID,
            $data->Name,
        );
    }
}
